packages/@guardian/source-foundations/src/typography/types.ts

```diff
@@ -48,12 +48,51 @@ type Categories = {
 
 export type Category = keyof Categories;
 
-export type AvailableFontsMapping = {
-	[cat in Category]: {
-		[fontWeight in FontWeight]?: FontWeightDefinition;
-	};
+export type AvailableFontWeights = {
+	[cat in Category]?: { [weight in FontWeight]?: boolean };
 };
 
+export type ItalicsAvailableForFontWeight = {
+	[cat in Category]?: { [weight in FontWeight]?: boolean };
+};
+
+export type TypographyOptions = {
+	lineHeight: LineHeight;
+	fontWeight: FontWeight;
+	fontStyle: Option<FontStyle>;
+	unit: ScaleUnit;
+};
+
+export type FontScaleArgs = Partial<
+	Pick<TypographyOptions, 'fontWeight' | 'lineHeight' | 'unit'>
+> & {
+	fontStyle?: FontStyle;
+};
+
+export type FontScaleFunction = (options?: FontScaleArgs) => TypographyStyles;
+
+// returns styles as a template literal
+export type FontScaleFunctionStr = (options?: FontScaleArgs) => string;
+
+export type TitlepieceFunctions = {
```

Comment:
Could we define this as a common type? ie:
```ts
type TypographyFunction<T> = {
	[key in keyof T]: FontScaleFunction;
};

export type HeadlineFunctions = TypographyFunction<HeadlineSizes>

export type TitlepieceFunctions = TypographyFunction<TitlepieceSizes>
```
and so on..

Reply:
nice, that's a lot cleaner

Reply:
https://github.com/guardian/source/pull/1525/files#diff-ecf8ea629817e204e3ed85acdc26e07fb6524496ff250307ddfede20929a16d4R82-R88

Reply:
Nicely done!